<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>RaytracerGui</class>
 <widget class="QMainWindow" name="RaytracerGui">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1561</width>
    <height>727</height>
   </rect>
  </property>
  <property name="mouseTracking">
   <bool>false</bool>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_13">
    <item>
     <widget class="QTabWidget" name="leftTabWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="cameraTab">
       <attribute name="title">
        <string>Camera</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_8">
        <item>
         <widget class="QGroupBox" name="dimensionsGroupBox">
          <property name="title">
           <string>Dimensions</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QGroupBox" name="captureGroup">
             <property name="title">
              <string>Capture</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout">
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout">
                <item>
                 <widget class="QSpinBox" name="captureDimensionsSpinX">
                  <property name="suffix">
                   <string> x</string>
                  </property>
                  <property name="minimum">
                   <number>16</number>
                  </property>
                  <property name="maximum">
                   <number>3840</number>
                  </property>
                  <property name="value">
                   <number>480</number>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QSpinBox" name="captureDimensionsSpinY">
                  <property name="suffix">
                   <string> y</string>
                  </property>
                  <property name="minimum">
                   <number>16</number>
                  </property>
                  <property name="maximum">
                   <number>2160</number>
                  </property>
                  <property name="value">
                   <number>300</number>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="outputDimensionsGroup">
             <property name="title">
              <string>Output</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_12">
              <item>
               <widget class="QRadioButton" name="matchCaptureSizeRadioBtn">
                <property name="text">
                 <string>Match capture size</string>
                </property>
                <property name="checked">
                 <bool>false</bool>
                </property>
                <attribute name="buttonGroup">
                 <string notr="true">outputDimensionsButtonGroup</string>
                </attribute>
               </widget>
              </item>
              <item>
               <widget class="QRadioButton" name="customOutputDimensionsRadioBtn">
                <property name="text">
                 <string>Use custom dimensions</string>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
                <attribute name="buttonGroup">
                 <string notr="true">outputDimensionsButtonGroup</string>
                </attribute>
               </widget>
              </item>
              <item>
               <widget class="QWidget" name="customOutputDimensionsWidget" native="true">
                <property name="enabled">
                 <bool>true</bool>
                </property>
                <layout class="QHBoxLayout" name="customOutputDimensionsLayout">
                 <item>
                  <widget class="QSpinBox" name="outputDimensionsSpinX">
                   <property name="suffix">
                    <string> x</string>
                   </property>
                   <property name="minimum">
                    <number>16</number>
                   </property>
                   <property name="maximum">
                    <number>3840</number>
                   </property>
                   <property name="value">
                    <number>960</number>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QSpinBox" name="outputDimensionsSpinY">
                   <property name="suffix">
                    <string> y</string>
                   </property>
                   <property name="minimum">
                    <number>16</number>
                   </property>
                   <property name="maximum">
                    <number>2160</number>
                   </property>
                   <property name="value">
                    <number>600</number>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="fullscreenButton">
             <property name="text">
              <string>Show fullscreen (Esc to exit)</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="fieldOfViewGroup">
          <property name="title">
           <string>Field of view</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_19">
           <item>
            <widget class="QSlider" name="fieldOfViewSlider">
             <property name="minimum">
              <number>10</number>
             </property>
             <property name="maximum">
              <number>100</number>
             </property>
             <property name="value">
              <number>40</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="tickPosition">
              <enum>QSlider::TicksBelow</enum>
             </property>
             <property name="tickInterval">
              <number>10</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="fieldOfViewSpin">
             <property name="suffix">
              <string>deg</string>
             </property>
             <property name="minimum">
              <number>10</number>
             </property>
             <property name="maximum">
              <number>100</number>
             </property>
             <property name="value">
              <number>40</number>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="cameraSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="sceneTab">
       <attribute name="title">
        <string>Scene</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_11">
        <item>
         <widget class="QTreeView" name="sceneTreeView"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QScrollArea" name="raytracedScrollView">
      <property name="mouseTracking">
       <bool>false</bool>
      </property>
      <property name="horizontalScrollBarPolicy">
       <enum>Qt::ScrollBarAsNeeded</enum>
      </property>
      <property name="widgetResizable">
       <bool>false</bool>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <widget class="QWidget" name="raytracedScollViewContents">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>978</width>
         <height>660</height>
        </rect>
       </property>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QTabWidget" name="rightTabWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="toolTip">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="postProdTab">
       <attribute name="title">
        <string>Post Prod</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_7">
        <item>
         <widget class="QGroupBox" name="lowpassFilterGroup">
          <property name="title">
           <string>Low-pass Filter</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <item>
            <widget class="QCheckBox" name="activateLowpassCheck">
             <property name="text">
              <string>Activate low-pass</string>
             </property>
             <property name="checked">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QWidget" name="lowpassWidget" native="true">
             <property name="enabled">
              <bool>true</bool>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_6">
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <layout class="QFormLayout" name="lowpassLayout">
                <item row="0" column="0">
                 <widget class="QLabel" name="lowpassSizeLabel">
                  <property name="text">
                   <string>Size: </string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <widget class="QComboBox" name="lowpassSizeCombo">
                  <property name="enabled">
                   <bool>true</bool>
                  </property>
                  <item>
                   <property name="text">
                    <string>3x3</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>5x5</string>
                   </property>
                  </item>
                 </widget>
                </item>
                <item row="1" column="0">
                 <widget class="QLabel" name="lowpassVarianceLabel">
                  <property name="text">
                   <string>Variance: </string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="1">
                 <widget class="QDoubleSpinBox" name="lowpassVarianceSpin">
                  <property name="enabled">
                   <bool>true</bool>
                  </property>
                  <property name="maximum">
                   <double>5.000000000000000</double>
                  </property>
                  <property name="singleStep">
                   <double>0.100000000000000</double>
                  </property>
                  <property name="value">
                   <double>0.500000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <widget class="QCheckBox" name="useAdaptativeFilteringCheck">
                <property name="text">
                 <string>Use adaptative filtering</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QWidget" name="adaptativeLayout" native="true">
                <property name="enabled">
                 <bool>true</bool>
                </property>
                <layout class="QHBoxLayout" name="horizontalLayout_11">
                 <property name="leftMargin">
                  <number>0</number>
                 </property>
                 <property name="topMargin">
                  <number>0</number>
                 </property>
                 <property name="rightMargin">
                  <number>0</number>
                 </property>
                 <property name="bottomMargin">
                  <number>0</number>
                 </property>
                 <item>
                  <widget class="QSlider" name="adaptativeFactorSlider">
                   <property name="enabled">
                    <bool>true</bool>
                   </property>
                   <property name="maximum">
                    <number>100</number>
                   </property>
                   <property name="value">
                    <number>100</number>
                   </property>
                   <property name="orientation">
                    <enum>Qt::Horizontal</enum>
                   </property>
                   <property name="tickPosition">
                    <enum>QSlider::TicksBelow</enum>
                   </property>
                   <property name="tickInterval">
                    <number>25</number>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLabel" name="adaptativeFactorLabel">
                   <property name="enabled">
                    <bool>true</bool>
                   </property>
                   <property name="minimumSize">
                    <size>
                     <width>45</width>
                     <height>0</height>
                    </size>
                   </property>
                   <property name="maximumSize">
                    <size>
                     <width>45</width>
                     <height>16777215</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>100%</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <widget class="QTableWidget" name="kernelTable">
                <property name="minimumSize">
                 <size>
                  <width>225</width>
                  <height>150</height>
                 </size>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>225</width>
                  <height>150</height>
                 </size>
                </property>
                <property name="verticalScrollBarPolicy">
                 <enum>Qt::ScrollBarAlwaysOff</enum>
                </property>
                <property name="horizontalScrollBarPolicy">
                 <enum>Qt::ScrollBarAlwaysOff</enum>
                </property>
                <attribute name="horizontalHeaderVisible">
                 <bool>false</bool>
                </attribute>
                <attribute name="horizontalHeaderDefaultSectionSize">
                 <number>45</number>
                </attribute>
                <attribute name="horizontalHeaderMinimumSectionSize">
                 <number>41</number>
                </attribute>
                <attribute name="verticalHeaderVisible">
                 <bool>false</bool>
                </attribute>
                <attribute name="verticalHeaderDefaultSectionSize">
                 <number>30</number>
                </attribute>
                <row>
                 <property name="text">
                  <string>x-2</string>
                 </property>
                </row>
                <row>
                 <property name="text">
                  <string>x-1</string>
                 </property>
                </row>
                <row>
                 <property name="text">
                  <string>x0</string>
                 </property>
                </row>
                <row>
                 <property name="text">
                  <string>x+1</string>
                 </property>
                </row>
                <row>
                 <property name="text">
                  <string>x+2</string>
                 </property>
                </row>
                <column>
                 <property name="text">
                  <string>y-2</string>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>y-1</string>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>y0</string>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>y+1</string>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>y+2</string>
                 </property>
                </column>
                <item row="0" column="0">
                 <property name="text">
                  <string/>
                 </property>
                 <property name="flags">
                  <set>NoItemFlags</set>
                 </property>
                </item>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="temperatureGroup">
          <property name="title">
           <string>Temperature</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_9">
           <item>
            <widget class="QSpinBox" name="temperatureSpin">
             <property name="buttonSymbols">
              <enum>QAbstractSpinBox::UpDownArrows</enum>
             </property>
             <property name="suffix">
              <string> K</string>
             </property>
             <property name="minimum">
              <number>2000</number>
             </property>
             <property name="maximum">
              <number>20000</number>
             </property>
             <property name="singleStep">
              <number>250</number>
             </property>
             <property name="value">
              <number>6500</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="temperatureDefaultBtn">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="sizePolicy">
              <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Default</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="luminosityGroup">
          <property name="title">
           <string>Luminosity</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_12">
           <item>
            <widget class="QSlider" name="luminositySlider">
             <property name="maximum">
              <number>100</number>
             </property>
             <property name="value">
              <number>50</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="tickPosition">
              <enum>QSlider::TicksBelow</enum>
             </property>
             <property name="tickInterval">
              <number>25</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="luminosityLabel">
             <property name="minimumSize">
              <size>
               <width>45</width>
               <height>0</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>45</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="text">
              <string>0.0</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="contrastGroup">
          <property name="title">
           <string>Contrast</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_10">
           <item>
            <widget class="QSlider" name="contrastSlider">
             <property name="maximum">
              <number>100</number>
             </property>
             <property name="value">
              <number>50</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="tickPosition">
              <enum>QSlider::TicksBelow</enum>
             </property>
             <property name="tickInterval">
              <number>25</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="contrastLabel">
             <property name="minimumSize">
              <size>
               <width>45</width>
               <height>0</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>45</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="text">
              <string>100%</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="saveButton">
          <property name="text">
           <string>Save Ouput Image</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="postProdSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="materialTab">
       <attribute name="title">
        <string>Material</string>
       </attribute>
      </widget>
     </widget>
    </item>
   </layout>
   <zorder>rightTabWidget</zorder>
   <zorder>raytracedScrollView</zorder>
   <zorder>leftTabWidget</zorder>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1561</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="outputDimensionsButtonGroup"/>
 </buttongroups>
</ui>
